File: backend/app/schemas/ai.py
from datetime import datetime
from decimal import Decimal
from typing import Any

from pydantic import BaseModel, ConfigDict


class ParsedResumeRead(BaseModel):
    resume_id: int
    candidate_id: int
    parsed_json: dict[str, Any]
    raw_text: str | None
    parse_status: str


class AIResumeAnalysisRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    candidate_id: int
    resume_id: int
    summary: str | None
    skills_json: list[Any] | None
    work_experience_summary: str | None
    project_experience_summary: str | None
    education_summary: str | None
    strengths_json: list[Any] | None
    risks_json: list[Any] | None
    interview_questions_json: list[Any] | None
    raw_response: dict[str, Any] | None
    model_provider: str | None
    model_name: str | None
    status: str
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class JobMatchScoreRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    job_id: int
    candidate_id: int
    application_id: int | None
    total_score: Decimal | None
    level: str | None
    dimension_scores_json: dict[str, Any] | None
    matched_points_json: list[Any] | None
    missing_points_json: list[Any] | None
    risk_points_json: list[Any] | None
    recommendation: str | None
    explanation: str | None
    raw_response: dict[str, Any] | None
    model_provider: str | None
    model_name: str | None
    status: str
    error_message: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/ai.ts
export interface ParsedResumeResult {
  resume_id: number;
  candidate_id: number;
  parsed_json: Record<string, unknown>;
  raw_text?: string;
  parse_status: string;
}

export interface AuditLog {
  id: number;
  user_id?: number | null;
  action: string;
  target_type?: string | null;
  target_id?: number | null;
  detail_json?: Record<string, unknown> | null;
  ip_address?: string | null;
  user_agent?: string | null;
  created_at: string;
  updated_at: string;
}

export interface WorkflowInterpretRequest {
  instruction: string;
  job_id?: number;
  candidate_id?: number;
  resume_id?: number;
}

export interface WorkflowInterpretResult {
  instruction: string;
  intent: string;
  confidence: number;
  matched_keywords: string[];
  required_inputs: string[];
  missing_inputs: string[];
  recommended_action: string;
  suggested_steps: string[];
  execution_hint: string;
  can_execute: boolean;
  job_id?: number | null;
  candidate_id?: number | null;
  resume_id?: number | null;
}

export interface AIResumeAnalysis {
  id: number;
  candidate_id: number;
  resume_id: number;
  summary?: string;
  skills_json?: unknown[];
  work_experience_summary?: string;
  project_experience_summary?: string;
  education_summary?: string;
  strengths_json?: unknown[];
  risks_json?: unknown[];
  interview_questions_json?: Array<{ type: string; question: string; focus: string }>;
  raw_response?: Record<string, unknown>;
  model_provider?: string;
  model_name?: string;
  status: string;
  error_message?: string;
  created_at: string;
  updated_at: string;
}

export interface JobMatchScore {
  id: number;
  job_id: number;
  candidate_id: number;
  application_id?: number;
  total_score?: string;
  level?: string;
  dimension_scores_json?: Record<string, number>;
  matched_points_json?: unknown[];
  missing_points_json?: unknown[];
  risk_points_json?: unknown[];
  recommendation?: string;
  explanation?: string;
  raw_response?: Record<string, unknown>;
  model_provider?: string;
  model_name?: string;
  status: string;
  error_message?: string;
  created_at: string;
  updated_at: string;
}
